expr_test.go

```diff
@@ -1811,6 +1811,97 @@ func TestEval_nil_in_maps(t *testing.T) {
 	})
 }
 
+// Test the use of env keyword.  Forms env[] and env[”] are valid.
+// The enclosed identifier must be in the expression env.
+func TestEnv_keyword(t *testing.T) {
+	env := map[string]interface{}{
+		"space test":                       "ok",
+		"space_test":                       "not ok", // Seems to be some underscore substituting happening, check that.
+		"Section 1-2a":                     "ok",
+		`c:\ndrive\2015 Information Table`: "ok",
+		"%*worst function name ever!!": func() string {
+			return "ok"
+		}(),
+		"env": func(string) string {
+			return "num"
+		},
+		"1":       "o",
+		"2":       "k",
+		"num":     10,
+		"my list": []int{1, 2, 3, 4, 5},
+		"MIN": func(a, b int) int {
+			if a < b {
+				return a
+			} else {
+				return b
+			}
+		},
+		"red":   "n",
+		"irect": "um",
+	}
+
+	// No error cases
+	var tests = []struct {
+		code string
+		want interface{}
+	}{
+		{"env['space test']", "ok"},
+		{"env['Section 1-2a']", "ok"},
+		{`env["c:\\ndrive\\2015 Information Table"]`, "ok"},
+		{"env['%*worst function name ever!!']", "ok"},
+		{"env('Confusing!')", "num"}, // not keyword, but some function named env
```

Comment:
This is a confusing part. Let's just forbid using the `env` keyword for anything else. 

We also can check this forbid rule on Compile step.

Reply:
I did this to preserve some backwards compatibility with variables called env, though it will be less confusing to disallow that case.  Note: other keywords may be used in this way - I can define a string called "len" or "filter" without error.

Reply:
> I can define a string called "len" or "filter" without error.

This is a thing to fix in the future. Disallow by default and only allow with `expr.AllowOverrideStandard()` option or something like this.

Reply:
A convenient place to check for keyword misuse is in types_table.go, when all map keys and struct tags are examined.